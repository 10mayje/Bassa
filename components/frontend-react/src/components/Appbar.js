import React from 'react';

//MUI imports
import {makeStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Avatar from '@material-ui/core/Avatar';
import TextField from '@material-ui/core/TextField';

//Logo import
import logo from '../logo.png';

const colours = {
  somewhat_brown: '#625042',
  white: '#fff'
}

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  avatar: {
    margin: 10,
  },
  bigAvatar: {
    margin: 10,
    width: 60,
    height: 60,
  },
}));

const BassaAppBar = (props) => {
  const classes = useStyles();

  if (!props.isloggedIn) {
    //If the user is not logged in, then return component for logged out user
  	return (
	  <div className={classes.root}>
	    <AppBar position="static" style={{background: colours.somewhat_brown}}>
	      <Toolbar>
		    <Avatar alt="Bassa" src={logo} className={classes.bigAvatar} />
	        <Typography variant="h4" color="inherit" className={classes.grow}>
	          Bassa
	        </Typography>
          <TextField 
            id="username"
            data-test="input-username"
            margin="normal"
            variant="outlined"
            style={{background:colours.white}}
            placeholder="Username" />&nbsp;&nbsp;
          <TextField 
            id="password"
            data-test="input-password"
            type="password"
            variant="outlined"
            margin="normal"
            style={{background:colours.white}}
            placeholder="Password" />&nbsp;&nbsp;
	        <Button size="large" variant="contained" color="primary">Login</Button>
	      </Toolbar>
	    </AppBar>
	  </div>
  	)
  }
  //Else return component for logged in user
  return (
  	<div className={classes.root}>
      <AppBar position="static" style={{background: colours.somewhat_brown}}>
        <Toolbar>
	      <Avatar alt="Bassa" src={logo} className={classes.avatar} />
          <Typography variant="h6" color="inherit" className={classes.grow}>
            Bassa
          </Typography>
          <Button data-test="button-dashboard" size="small" color="inherit">Dashboard</Button>
          <Button data-test="button-admin" size="small" color="inherit">Admin</Button>
          <Button data-test="button-logout" size="small" color="inherit">Logout</Button>
        </Toolbar>
      </AppBar>
    </div>
  )
}

export default BassaAppBar;