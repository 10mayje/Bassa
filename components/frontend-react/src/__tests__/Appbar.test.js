import React from 'react';
import { shallow } from 'enzyme';
import Appbar from '../components/Appbar';

const setupComponent = props => {
  const component = shallow(<Appbar {...props} />);
  return component;
}

const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
}

describe('Appbar Component Testing', () => {
  it('should match with its snapshot', () => {
    const component = setupComponent()
    expect(component).toMatchSnapshot();
  });
  describe('User is logged in', () => {
    it('should display logged in view', () => {
      const component = setupComponent({isloggedIn: true});
      const buttonLogout = findByTestAttr(component, 'button-logout');
      const buttonDashboard = findByTestAttr(component, 'button-dashboard');
      const buttonAdmin = findByTestAttr(component, 'button-admin');
      
      expect(buttonLogout.props().children).toBe('Logout');
      expect(buttonDashboard.props().children).toBe('Dashboard');
      expect(buttonAdmin.props().children).toBe('Admin');
    });
  });

  describe('User is logged out', () => {
    it('should display logged out view', () => {
      const component = setupComponent({isloggedIn: false});
      const inputUsername = findByTestAttr(component, 'input-username');
      const inputPassword = findByTestAttr(component, 'input-password');
      
      expect(inputUsername.props().id).toBe('username');
      expect(inputPassword.props().id).toBe('password');
    })
  })
});