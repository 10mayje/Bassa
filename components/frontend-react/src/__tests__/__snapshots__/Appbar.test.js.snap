// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appbar Component Testing should match with its snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
    style={
      Object {
        "background": "#625042",
      }
    }
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Avatar))
        alt="Bassa"
        className="makeStyles-bigAvatar-5"
        src="logo.png"
      />
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-grow-2"
        color="inherit"
        variant="h4"
      >
        Bassa
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(TextField))
        data-test="input-username"
        id="username"
        margin="normal"
        placeholder="Username"
        style={
          Object {
            "background": "#fff",
          }
        }
        variant="outlined"
      />
        
      <WithStyles(ForwardRef(TextField))
        data-test="input-password"
        id="password"
        margin="normal"
        placeholder="Password"
        style={
          Object {
            "background": "#fff",
          }
        }
        type="password"
        variant="outlined"
      />
        
      <WithStyles(ForwardRef(Button))
        color="primary"
        size="small"
        variant="contained"
      >
        Login
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
